lib/lz4.c

```diff
@@ -1727,7 +1727,16 @@ LZ4_decompress_generic(
 
                 /* Fastpath check: Avoids a branch in LZ4_wildCopy32 if true */
                 if (!(dict == usingExtDict) || (match >= lowPrefix)) {
+#if defined(__aarch64__) && !defined(__clang__)
+                    if (offset >= 16) {
+                        memcpy(op, match, 16);
+                        memcpy(op+16, match+16, 2);
+                        op += length;
+                        continue;
+                     } else if (offset >= 8) {
```

Comment:
I'm surprised that adding a branch ends up being so beneficial for `aarch64`.
This is in contrast with expectation, since each added branch adds a risk of pipeline flush.
That being said, your patch is backed by solid benchmark results, so I'll investigate.

Reply:
I am not sure the root cause till now, I am not sure it is related with memcpy performance.
It seems like the more data copied in this condition check, the more performance binefit we could got.